### DISCUSSION TITLE
Retag existing Formula

### DISCUSSION THREAD

**Discussion by enrichman** [Homebrew/discussions#5515] [Tap maintenance and Homebrew development]
We have tagged a wrong version and we were thinking about deleting the tags, fix the error and retagging.

Is it possible? What is going to happen?

If some users will keep the old "wrong" version it should be fine. We can eventually notify them in the release notes.

**Comment by SMillerDev** [comment]
The [git manual](https://git-scm.com/docs/git-tag#_on_re_tagging) says re-tagging is "the insane thing" to do.

A much better option when you change things after releasing something is to increase the version number and release a new version.

**Reply by enrichman** [reply]
Yeah, I know. Unfortunately this is not an option in our case, because the tag needs to match other artifacts. So I was just trying to understand what will happen on the Brew side.

IIRC new installation will get the new binary, while the old one will stick with the wrong.

**Reply by SMillerDev** [reply]
No, builds in Homebrew will fail and someone will need to go investigate if your codebase was trying to one-up XZ or if you decided to do what git explicitly calls out as a bad idea.

By far the best way is to release a new version, otherwise your version numbers are meaningless too. For example, 2.4 today could be a different piece of software than 2.4 tomorrow.
